src/core/public/core_system.ts

```diff
@@ -205,7 +205,7 @@ export class CoreSystem {
       const docLinks = this.docLinks.start({ injectedMetadata });
       const http = await this.http.start();
       const workspaces = await this.workspaces.start();
-      const savedObjects = await this.savedObjects.start({ http });
+      const savedObjects = await this.savedObjects.start({ http, workspaces });
```

Comment:
I tend to design the core `saved_objects` service as much independent as possible, so that it won't rely on a specific feature, such as `workspace`.

What do you think? @SuZhou-Joe @wanglam 

```typescript
// src/core/public/saved_objects/saved_objects_client.ts
export class SavedObjectsClient {
  defaultNamespace = ''

  setDefaultNamespace(ns: string) {
    this.defaultNamespace = ns
  }
}

// src/core/public/core_system.ts
const workspaces = await this.workspaces.start({ savedObjects });

// src/core/public/workspace/workspaces_client.ts
export class WorkspacesClient {
  constructor(http, savedObjects) {
    this.currentWorkspace$.subscribe((current) => {
      if (current) {
        savedObject.setDefaultNamespace(current.id)
      }
    })
  }
}
```

Reply:
Sounds reasonable to me, will change this logic.